lib/AudioManager.dart

```diff
@@ -81,8 +82,16 @@ class AudioManager {
   }
 
   Future<int> fetchAudioDuration(String audioFile) async {
-    return await audioPlayer.getDuration() ?? 0; // Returns duration in milliseconds
-  }
+  // Load the audio file without playing it
+  await audioPlayer.setSource(AssetSource(audioFile));
+  
+  // Wait briefly to ensure the audio file is loaded
+  await Future.delayed(Duration(milliseconds: 100));
+
+  // Now retrieve the duration
+  int? duration = await audioPlayer.getDuration();
```

Comment:
Code still doesn't compile. Please ensure you are using the right types for variables